### PR TITLE
tldr translation badges

### PR DESCRIPTION
**Pull request by sbrl**
This is a request for comment / work in progress. I've implemented a quick Node.js script that generates translation progress badges as SVG images (because it's sooo much easier than trying to render a PNG lol).

Here's what it looks like so far:

Selection_2021-03-10_02:29:41_001_71edf24

_(screenshot because inkscape doesn't like `rgba()` in the colour of a fill - ignore the white stripe at the right hand side)_

The colour coding on the percentage cab be either green, yellow, orange, or red. Currently I have it set to 90%+ for green, 50%+ for yellow, 10%+ for orange - otherwise red - though if we were to actually use this we'd perhaps want to discuss what values would be appropriate lol :P

Implementation wise, it uses no npm packages (though it could have done easily) and is designed to be as minimal as possible to make it as easy to run as possible.

This is more an exploration and an experiment, but I'd appreciate comments + feedback.

<details><summary>Example SVG output</summary>
<pre class="language-svg"><code>&lt;svg xmlns=&quot;http://www.w3.org/2000/svg&quot; width=&quot;150&quot; height=&quot;590&quot; version=&quot;1.1&quot;&gt;	&lt;rect x=&quot;0&quot; y=&quot;0&quot; width=&quot;150&quot; height=&quot;590&quot; fill=&quot;#203050&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;20&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;bold&quot; fill=&quot;#97d6d2&quot;&gt;tldr-pages&lt;/text&gt;
	&lt;text x=&quot;10&quot; y=&quot;40&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;translation progress&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;50&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(128, 63%, 45%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;50&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;64&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;en&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;64&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;100.00%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;70&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;84&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;bs&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;84&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.44%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;90&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;90&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;104&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;da&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;104&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.04%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;110&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;124&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;de&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;124&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;6.70%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;130&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;130&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;144&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;es&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;144&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;5.46%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;150&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;164&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;fa&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;164&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.84%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;170&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;170&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;184&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;fr&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;184&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;6.57%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;190&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;204&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;hbs&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;204&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.44%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;210&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;210&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;224&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;hi&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;224&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.49%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;230&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;244&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;id&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;244&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;2.13%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;250&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(32, 93%, 48%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;250&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;264&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;it&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;264&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;14.56%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;270&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;284&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;ja&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;284&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;1.02%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;290&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;290&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;304&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;ko&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;304&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;7.94%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;310&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;324&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;ml&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;324&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.22%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;330&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;330&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;344&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;nl&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;344&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;1.38%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;350&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;364&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;no&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;364&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.44%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;370&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;370&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;384&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;pl&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;384&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;2.09%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;390&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;404&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;pt_BR&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;404&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;6.70%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;410&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;410&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;424&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;pt_PT&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;424&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.40%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;430&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;444&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;ru&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;444&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;1.20%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;450&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;450&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;464&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;sv&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;464&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;1.20%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;470&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;484&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;ta&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;484&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;1.33%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;490&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;490&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;504&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;th&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;504&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.04%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;510&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;524&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;tr&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;524&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;2.00%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;530&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(32, 93%, 48%)&quot; /&gt;	&lt;rect x=&quot;0&quot; y=&quot;530&quot; width=&quot;150&quot; height=&quot;20&quot; fill=&quot;rgba(200, 200, 200, 0.25)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;544&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;zh&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;544&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;14.91%&lt;/text&gt;
	&lt;rect x=&quot;64&quot; y=&quot;550&quot; width=&quot;86&quot; height=&quot;20&quot; fill=&quot;hsl(10, 72%, 58%)&quot; /&gt;	&lt;text x=&quot;10&quot; y=&quot;564&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;zh_TW&lt;/text&gt;
	&lt;text x=&quot;74&quot; y=&quot;564&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#333333&quot;&gt;0.44%&lt;/text&gt;
	&lt;text x=&quot;10&quot; y=&quot;584.4&quot; font-size=&quot;12&quot; font-family=&quot;sans-serif&quot; font-weight=&quot;normal&quot; fill=&quot;#97d6d2&quot;&gt;Updated 2021-03-10&lt;/text&gt;
	&lt;path d=&quot;M 0 50 l 150 0 M 64 50 L 64 570 M 0 570 L 150 570&quot; stroke=&quot;#4d9182&quot; stroke-width=&quot;4px&quot; /&gt;
&lt;/svg&gt;</code></pre>

### CODE REVIEW

File: scripts/badge/index.mjs [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env node
+"use strict";
```
Comment (navarroaxel) [inline scripts/badge/index.mjs:2]: I think the `use strict` is not needed, specially in a `.mjs` file.

File: scripts/badge/index.mjs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+function get_date() {
+	let d = new Date();
+	return d.toISOString().split("T")[0];
+}
+
+(async () => {
+    "use strict";
+    
+	console.log(`>>> Counting pages [ 1 / 2 ]`);
+    let pages = await count_pages(dirpath);
```
Comment (navarroaxel) [inline scripts/badge/index.mjs:58]: Why this extra indent?
```suggestion
  let pages = await count_pages(dirpath);
```

File: scripts/badge/index.mjs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+if(typeof filepath_output !== "string") {
+	console.error(`tldr translations badge maker
+
+Usage:
+	scripts/badge/index.mjs <path_to_repo_root> <path_to_output_file.svg>`);
+	process.exit();
+}
+
+function get_colour_percent(percent) {
```
Comment (navarroaxel) [inline scripts/badge/index.mjs:39]: please use arrow functions
```suggestion
const get_colour_percent = (percent) => {
```

File: scripts/badge/index.mjs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	font_size: 20,
+	font_family: "sans-serif",
+	threshold: {
+		green: 0.9,
+		yellow: 0.5,
+		orange: 0.1
+	}
+};
+
+if(typeof filepath_output !== "string") {
```
Comment (navarroaxel) [inline scripts/badge/index.mjs:31]:
```suggestion
if (typeof filepath_output !== "string") {
```

File: scripts/badge/LowLevelWriter.mjs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	
+	write(chunk) {
+		return new Promise((resolve, reject) => {
+			this.is_writing = true;
+			this.emit("write_start");
+			if(!Buffer.isBuffer(chunk))
+				chunk = Buffer.from(chunk.toString(), "utf8");
+			
+			fs.write(this.fd, chunk, (err, bytes_written) => {
+				if(err) {
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:45]:
```suggestion
				if (err) {
```

File: scripts/badge/LowLevelWriter.mjs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+					reject(err);
+					return false;
+				}
+				if(bytes_written != chunk.length) {
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:49]:
```suggestion
				if (bytes_written != chunk.length) {
```

File: scripts/badge/LowLevelWriter.mjs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+ * Imported from our personal massCode snippets db, and written
+ * to fix the most frustrating bug in my PhD data processing code....
+ * @extends EventEmitter
+ */
+class LowLevelWriter extends EventEmitter {
+	constructor() {
+		super();
+		
+		this.fd = null;
+		this.is_writing = false;
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:18]: As a JS dev :disappointed: the standard in JS is `camelCase`. Why you choose snake_case?
```suggestion
		this.isWriting = false;
```

**Review by bl-ue — CHANGES_REQUESTED** [review]
Wow, this is really nice @sbrl!

Since this is actual code, I recommend that we use something like [eslint](https://eslint.org) to settle the differences in maintainers' formatting preferences.  We're already doing this in the [node client](https://github.com/tldr-pages/tldr-node-client/blob/b206ac7de17f130dd9d874b5304b47f0165b39d3/.eslintrc.json), and the [linter](https://github.com/tldr-pages/tldr-lint/blob/main/.eslintrc.json).

Also, skip @navarroaxel's formatting if you do this because eslint will alert you about all formatting and style errors, and others 😉

**Comment by navarroaxel** [comment]
> Also, I'd like to ask: why are you using `.mjs` files?

I think he tried to avoid the definition of a `package.json` with `type: "module"` using `.mjs` directly, that's allow to use ES modules, instead of `require` from `commonjs`.

**Comment by bl-ue** [comment]
But why not use `package.json` and `require()`?

File: scripts/badge/index.mjs (hunk 1 above)
Comment (sbrl) [reply scripts/badge/index.mjs:2]: Oh? I wasn't aware of that?

File: scripts/badge/index.mjs (hunk 2 above)
Comment (sbrl) [reply scripts/badge/index.mjs:58]: typo :P

File: scripts/badge/LowLevelWriter.mjs (hunk 7 above)
Comment (sbrl) [reply scripts/badge/LowLevelWriter.mjs:18]: I was not aware of this. I've always used snake_case? is this a particular issue here?

**Comment by bl-ue** [comment]
All good points.  I think I agree with all of them. 👍🏻

**Comment by navarroaxel** [comment]
> @navarroaxel: *she

@sbrl My apologies, all this months and I didn't ask for your real name :grimacing:

**Comment by sbrl** [comment]
@bl-ue unfortunately despite the name `eslint` doesn't seem to like `import` very much:

```
$ /tmp/node_modules/.bin/eslint --no-eslintrc /tmp/tldr/scripts/badge/index.mjs 

/tmp/tldr/scripts/badge/index.mjs
  4:1  error  Parsing error: The keyword 'import' is reserved

✖ 1 problem (1 error, 0 warnin
/tmp/tldr/scripts/badge/index.mjs
  4:1  error  Parsing error: The keyword 'import' is reserved

✖ 1 problem (1 error, 0 warnings)
```

I feel as though the name `eslint` is misleading.

...I feel as though the name

**Comment by bl-ue** [comment]
Perhaps [this](https://stackoverflow.com/questions/36002226/parsing-error-the-keyword-import-is-reserved-sublimelinter-contrib-eslint) would help? Also, [this page](https://eslint.org/docs/rules/no-useless-rename#disallow-renaming-import-export-and-destructured-assignments-to-the-same-name-no-useless-rename) really is implying that `import`/`export` are allowed somehow.

**Comment by sbrl** [comment]
Hrm, that's not really helping @bl-ue. Those SO answers are talking about using Visual Studio Code, which I'm not using. I gave it a quick try with the `eslint` CLI. Other answers suggest edits to a `.eslintrc`, but I really don't want to go to the trouble of writing one just for this one-off occasion.

**Comment by bl-ue** [comment]
Well @sbrl we probably should as we may get more JS scripts and we already use `.eslintrc.json` in our other node repos:

- https://github.com/tldr-pages/tldr-node-client/blob/master/.eslintrc.json
- https://github.com/tldr-pages/tldr-lint/blob/main/.eslintrc.json

**Review by marchersimon — APPROVED** [review]
I don't know any JavaScript, but here's my approval since I don't think another script would hurt.

File: scripts/badge/LowLevelWriter.mjs [hunk 8]
```diff
@@ -0,0 +1,89 @@
+"use strict";
+
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:2]: I think the script filenames should be in camelCase.js
```suggestion
// SPDX-License-Identifier: MIT
```

File: scripts/badge/LowLevelWriter.mjs [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+	constructor() {
+		super();
+		
+		this.fd = null;
+		this.is_writing = false;
+	}
+	
+	open(filename) {
+		return new Promise((resolve, reject) => {
+			if(this.fd !== null) {
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:23]:
```suggestion
			if (this.fd !== null) {
```

File: scripts/badge/LowLevelWriter.mjs [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+				reject("InvalidOperationError: Can't open multiple files on a single LowLevelWriter.");
+				return false;
+			}
+			
+			fs.open(filename, "w", (error, fd) => {
+				if(error) reject(error);
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:29]:
```suggestion
				if (error) {
				  reject(error);
				  return;
				}
```

File: scripts/badge/LowLevelWriter.mjs [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+		
+		this.fd = null;
+		this.is_writing = false;
+	}
+	
+	open(filename) {
+		return new Promise((resolve, reject) => {
+			if(this.fd !== null) {
+				reject("InvalidOperationError: Can't open multiple files on a single LowLevelWriter.");
+				return false;
```
Comment (navarroaxel) [inline scripts/badge/LowLevelWriter.mjs:25]: why?
```suggestion
				return;
```

File: scripts/badge/LowLevelWriter.mjs (hunk 7 above)
Comment (navarroaxel) [reply scripts/badge/LowLevelWriter.mjs:18]: No, just the convention.

**Review by navarroaxel — CHANGES_REQUESTED** [review]

**Comment by Waples** [comment]
Removing this PR. Reason: Stale/dead since 2021

**Pull request closed without merging**